src/pybind/mgr/dashboard_v2/settings.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import errno
+import inspect
+from six import add_metaclass
+
+from . import mgr
+
+
+class Options(object):
+    """
+    If you need to store some configuration value please add the config option
+    name as a class attribute to this class.
+
+    Example::
+
+        GRAFANA_API_HOST = ('localhost', str)
+        GRAFANA_API_PORT = (3000, int)
+    """
+    pass
+
+
+class SettingsMeta(type):
+    def __getattr__(cls, attr):
+        if hasattr(Options, attr):
```

Comment:
`hasattr` looks like this:
```python
def __hasattr__(self, name):
    try:
        getattr(self, name)
        return True
    except AttributeError: 
        return False
```
No need to duplicate catching `AttributError` here. What about simplifying `SettingsMeta.__getattr__` to:
```python
def __getattr__(cls, attr):
    default, stype = getattr(Options, attr)
    return stype(mgr.get_config(attr, default))
```
?

Reply:
sounds good